### DISCUSSION TITLE
File Transfer suggested enhancements

### DISCUSSION THREAD

**Discussion by Reikooters** [rustdesk/rustdesk#2065] [General]
Thanks to the team on the work you've done so far to get RustDesk to this point. 👍 

Here are my suggested enhancements for File Transfer window/dialog on Desktop (tested on Windows 10)

**1) Typing to select a file/folder**
In the list box showing the files/folders at each side, typing multiple letters when the listbox is focused should move the highlight/cursor to a file/folder with a name beginning with those characters, similar to how it works in Windows Explorer.

Example: Imagine a folder with 3 files in it:

-  abc.txt
-  home.txt
-  shared.txt

In Windows Explorer, typing "sha" will cause the **sha**red.txt file to become highlighted, because the filename begins with these 3 characters.

In the current RustDesk File Transfer dialog, the cursor will first go to **s**hared.txt, then the next keystroke will immediately move the cursor to **h**ome.txt, then the 3rd keystroke will immediately move the cursor to **a**bc.txt - currently only allowing one keypress before it moves to another file starting with the next key.

**2) Ability to use native "browse" for a folder to select the folder**
i.e. a Browse button next to the folder path which would open the native Windows folder select to navigate to the folder quickly - not sure how this might work on other devices.

**3) Warn when sending/receiving a file that exists at the destination**
Currently, for both sending or receiving a file, if the filename exists at the destination, it is overwritten without any warning. The application should warn you for files that exist and allow you to cancel/skip or overwrite. Deleting a file currently has a confirmation, but overwrite does not.

**4) Add a button or a way to clear transfer history shown at the right side**
When transferring many files from a folder with a large number of files, it creates many log entries in the history shown at the right side. It seems currently to clear this you need to click the X one by one to clear each log entry. Having a way to clear this history (e.g. a clear button above the list) would be beneficial in these cases.

**5) Add timestamp to transfer history, and sort results by newest first**
Maybe it is simply my preference, but I also think that the newest entries should appear at the top of the list instead of the bottom. This also makes the history easier to read when the history becomes longer, since the newest entry is always at the top.

A timestamp of when the history record was created (even if just the time without the date to save screen space) would be good too. That way if you are working on a file and updating/overwriting it multiple times, you can use the history to tell that you have recently sent the file based on the timestamp, without the need to go and check the file timestamp in Windows Explorer at either the remote/local side.

**6) (not part of the File Transfer dialog) Allow copy/paste a file from the client PC directly into a folder on the remote PC**
I'm guessing this one is more difficult to implement but it would be a "nice to have" feature. Basically, allow copying a file or folder(s) from Windows Explorer in the client PC, then paste them directly into a folder in Windows Explorer on the remote PC. When pressing Ctrl+V/right click > Paste/etc at the destination, this would automatically start a file transfer of the copied files from the appropriate source/to the appropriate destination, whether it's to/from the local/remote machine.

Some other remote desktop software I've tried supports this (TeamViewer, AnyDesk, Windows RDP) and some others do not (RemotePC).

**Comment by rustdesk** [comment]
PR welcome

**Comment by ROdoka** [comment]
I just found out about RustDesk and I am amazed already, because RustDesk is way faster than AD which I want to get rid of because of its nag screens and because it's also less userfriendly :)


- Delete key should delete the file(s), and if more than one file is selected it should delete all of them without asking for each file

- going up in folders only goes to the root of the drive, but not higher (the next step would usually show the drives)

- no direct file transfer button

- point 6) above would rock

- why does it need to open another connection for file transfer? I have to enter the password again when a connection to the remote system is already open

- maybe add compression to file transfer (the commercial TV for example has this IIRC)

anyway, keep up the good work, RustDesk rocks already :)

**Comment by rustdesk** [comment]
@Heap-Hop you can summerize, we can accept some which make sense from our view.
